key_management/src/keystore.rs

```diff
@@ -113,127 +117,387 @@ pub trait KeyStore {
     fn remove(&mut self, key: String) -> Result<KeyInfo, Error>;
 }
 
-#[derive(Default, Clone, PartialEq, Debug, Eq)]
-pub struct MemKeyStore {
-    pub key_info: HashMap<String, KeyInfo>,
+/// KeyStore struct, this contains a HashMap that is a set of KeyInfos resolved by their Address
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct KeyStore {
+    key_info: HashMap<String, KeyInfo>,
+    persistence: Option<PersistentKeyStore>,
+    encryption: Option<EncryptedKeyStore>,
 }
 
-impl MemKeyStore {
-    /// Return a new empty KeyStore
-    pub fn new() -> Self {
-        MemKeyStore {
-            key_info: HashMap::new(),
-        }
-    }
+pub enum KeyStoreConfig {
+    Memory(),
```

Comment:
Nit: we don't need the `()`

Reply:
Yeah, that's fair. I just figured it looked more consistent that way.

Reply:
Refactored.